galley/pkg/config/util/kuberesource/resources.go

```diff
@@ -16,19 +16,26 @@ package kuberesource
 
 import (
 	"istio.io/istio/galley/pkg/config/meta/schema"
+	"istio.io/istio/galley/pkg/config/meta/schema/collection"
+	"istio.io/istio/galley/pkg/config/processing/transformer"
 	"istio.io/istio/galley/pkg/config/source/kube/rt"
 	"istio.io/istio/galley/pkg/source/kube/builtin"
 )
 
 // DisableExcludedKubeResources is a helper that filters a KubeResources list to disable some resources
-// Behaves in the same way as existing logic:
+// The first filter behaves in the same way as existing logic:
 // - Builtin types are excluded by default.
 // - If ServiceDiscovery is enabled, any built-in type should be readded.
-func DisableExcludedKubeResources(input schema.KubeResources, excludedResourceKinds []string, enableServiceDiscovery bool) schema.KubeResources {
+// In addition, any resources not needed as inputs by the specified collections are disabled
+func DisableExcludedKubeResources(krs schema.KubeResources, xformProviders transformer.Providers,
```

Comment:
`input` is more readable than `krs` IMO. You could also probably just call `xFormProvider` `providers`.

Reply:
I didn't like `input` because the function has quite a few input values now and I thought it was getting confusing. Switched to `resources`. Also switched `xformProviders` to `providers`.

I'm still trying to get the hang of Go style here, when it makes sense to use descriptive names vs minimal names.

Reply:
`resources` and `providers` sounds good, but I don't see that change reflected in the last commit.